### PR TITLE
Consistent log messages

### PR DESCRIPTION
**Pull request by adams85**
### Describe the purpose of your pull request

Revises log messages to make them consistent with other SDKs and also adds event ids for easier log event identification.

Note: The current logging abstraction (`Logger` abstract class) doesn't support structured logging, so it can't be done without introducing BCs. If we want to enable it, I suggest following the solution used in the .NET SDK.

### Requirement checklist (only if applicable)

- [ ] I have covered the applied changes with automated tests.
- [ ] I have executed the full automated test set against my changes.
- [ ] I have validated my changes against all supported platform versions.
- [ ] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

File: lib/src/configcat_client.dart [hunk 1]
```diff
@@ -95,17 +95,17 @@ class ConfigCatClient {
...
         hooks.invokeFlagEvaluated(
             EvaluationDetails.makeError(key, defaultValue, err, evalUser));
         return defaultValue;
       }
       final setting = result.settings[key];
       if (setting == null) {
         final err =
-            'Value not found for key $key. Here are the available keys: ${result.settings.keys.join(', ')}';
-        _errorReporter.error(err);
+            'Failed to evaluate setting \'$key\' (the key was not found in config JSON). Returning the `defaultValue` parameter that you specified in your application: \'$defaultValue\'. Available keys: ${result.settings.keys.join(', ')}.';
```
Comment (adams85) [inline lib/src/configcat_client.dart]: Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`

File: lib/src/configcat_client.dart [hunk 2]
```diff
@@ -146,8 +146,8 @@ class ConfigCatClient {
...
```
Comment (adams85) [inline lib/src/configcat_client.dart]: Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`

File: lib/src/configcat_client.dart [hunk 3]
```diff
@@ -210,22 +210,22 @@ class ConfigCatClient {
...
+        _errorReporter.error(1000,
+            'Config JSON is not present. Returning the `defaultVariationId` parameter that you specified in your application: \'$defaultVariationId\'.');
         return defaultVariationId;
       }
       final setting = result.settings[key];
       if (setting == null) {
-        _errorReporter.error(
-            'Variation ID not found for key $key. Here are the available keys: ${result.settings.keys.join(', ')}');
+        _errorReporter.error(1001,
+            'Failed to evaluate setting \'$key\' (the key was not found in config JSON). Returning the `defaultVariationId` parameter that you specified in your application: \'$defaultVariationId\'. Available keys: ${result.settings.keys.join(', ')}.');
```
Comment (adams85) [inline lib/src/configcat_client.dart]: Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`

File: lib/src/fetch/config_fetcher.dart [hunk 4]
```diff
@@ -185,17 +185,17 @@ class ConfigFetcher implements Fetcher {
...
         final error =
-            'Unexpected HTTP response was received: ${response.statusCode} ${response.statusMessage}';
-        _errorReporter.error(error);
+            'Unexpected HTTP response was received while trying to fetch config JSON: ${response.statusCode} ${response.statusMessage}';
+        _errorReporter.error(1101, error);
         return FetchResponse.failure(error, true);
       }
     } catch (e, s) {
-      _errorReporter.error('Exception occurred during fetching.', e, s);
+      _errorReporter.error(1103, 'Unexpected error occurred while trying to fetch config JSON.', e, s);
```
Comment (adams85) [inline lib/src/fetch/config_fetcher.dart]: Request timeout (1102, `Request timed out while trying to fetch config JSON. Timeout value: {TIMEOUT}`) is not handled here.  I'm not sure it's an oversight or is just not possible in Dart.

**Review by z4kn4fein — APPROVED** [review]

**Pull request merged**